src/WebJobs.Extensions.Http/ClaimsPrincipalBindingProvider.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// This provider provides a binding to Type <see cref="ClaimsPrincipal"/>.
+    /// </summary>
+    internal class ClaimsPrincipalBindingProvider : IBindingProvider
+    {
+        private static Task<IBinding> nullBinding = Task.FromResult<IBinding>(null);
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsPrincipal))
+            {
+                return nullBinding;
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                IReadOnlyDictionary<string, object> bindingData = context.BindingData;
+                return Task.FromResult<IValueProvider>(new ClaimsPrincipalValueProvider(ClaimsPrincipalHelper.FromBindingData(bindingData)));
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                throw new NotImplementedException("This method does not provide the necessary binding data.");
```

Comment:
>NotImplementedException [](start = 26, length = 23)

Can you add a comment about when this overload would get called, and why it's safe for us to assume that it won't ever get called?

Reply:
After talking with Mike about this, it appears this is used when a parameter of the type ClaimsPrincipal is provided in a programatic call (i.e. via a normal WebJobs). We technically don't need an implementation, but I can right a simple one just to be safe.

---
In reply to: [174631670](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/373#discussion_r174631670) [](ancestors = 174631670)